pkg/localclusterimport/local_cluster_import_operations.go

```diff
@@ -0,0 +1,164 @@
+package localclusterimport
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+//go:generate mockgen -source=local_cluster_import_operations.go -package=localclusterimport -destination=local_cluster_import_operations_mocks.go
+type ClusterImportOperations interface {
+	GetAgentClusterInstall(namespace string, name string) (*hiveext.AgentClusterInstall, error)
+	CreateAgentClusterInstall(agentClusterInstall *hiveext.AgentClusterInstall) error
+	GetNamespace(name string) (*v1.Namespace, error)
+	GetSecret(namespace string, name string) (*v1.Secret, error)
+	CreateSecret(namespace string, secret *v1.Secret) (*v1.Secret, error)
+	GetClusterVersion(name string) (*configv1.ClusterVersion, error)
+	CreateClusterImageSet(clusterImageSet *hivev1.ClusterImageSet) (*hivev1.ClusterImageSet, error)
+	CreateClusterDeployment(clusterDeployment *hivev1.ClusterDeployment) (*hivev1.ClusterDeployment, error)
+	GetNodes() (*v1.NodeList, error)
+	GetNumberOfControlPlaneNodes() (int, error)
+	GetDNS(name string) (*configv1.DNS, error)
+}
+
+type Local_cluster_import_operations struct {
+	context         context.Context
+	apiReader       client.Reader
+	cachedApiClient client.Writer
+	log             *logrus.Logger
+}
+
+func NewLocalClusterImportOperations(apiReader client.Reader, cachedApiClient client.Writer, log *logrus.Logger) Local_cluster_import_operations {
+	return Local_cluster_import_operations{context: context.TODO(), apiReader: apiReader, cachedApiClient: cachedApiClient, log: log}
+}
+
+func (o *Local_cluster_import_operations) GetAgentClusterInstall(namespace string, name string) (*hiveext.AgentClusterInstall, error) {
+	agentClusterInstall := &hiveext.AgentClusterInstall{}
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, agentClusterInstall)
+	if err != nil {
+		return nil, err
+	}
+	return agentClusterInstall, nil
+}
+
+func (o *Local_cluster_import_operations) CreateAgentClusterInstall(agentClusterInstall *hiveext.AgentClusterInstall) error {
+	return o.cachedApiClient.Create(o.context, agentClusterInstall)
+}
+
+func (o *Local_cluster_import_operations) GetNamespace(name string) (*v1.Namespace, error) {
+	nsList := &v1.NamespaceList{}
+	err := o.apiReader.List(o.context, nsList)
```

Comment:
isn't there a get to a namespace? why do you need to do a list?

Reply:
I found that a call to get the namespace did not return the namespace